docs-src/cli/README.md

```diff
@@ -0,0 +1,234 @@
+For example, to disable the search attribute cache, run:
+
+```bash
+temporal server start-dev --dynamic-config-value system.forceSearchAttributesCacheRefreshOnRead=true
+```
+
+This setting makes created search attributes immediately available for use.
+
+## Auto-completion
+
+The Temporal CLI has the capability to auto-complete commands.
+
+Running `temporal completion SHELL` will output the related completion SHELL code.
+
+### zsh auto-completion
+
+<!-- TODO: add more information about zsh to make comparable to bash section -->
+
+Add the following code snippet to your `~/.zshrc` file:
+
+```sh
+source <(temporal completion zsh)
+```
+
+If you're running auto-completion from the terminal, run the command below:
+
+```sh
+echo 'source <(temporal completion zsh)' >> ~/.zshrc
+```
+
+After setting the variable, run:
+
+`source ~/.zshrc`.
+
+### Bash auto-completion
+
+Bash auto-completion relies on `bash-completion`.
+
+Install the software with the steps provided [here](https://github.com/scop/bash-completion#installation), or use your preferred package manager on your operating system.
+
+#### Linux installation
+
+Use any of the following package managers to install `bash-completion`:
+`apt install bash-completion`
+`pacman -S bash-completion`
+`yum install bash-completion`
+
+Verify that `bash-completion` is installed by running `type _init_completion`.
+
+To install the software on Alpine Linux, run:
+
+```bash
+apk update
+apk add bash-completion
+source /etc/profile.d/bash_completion.sh
+```
+
+Finally, enable completion for Temporal by adding the following code to your bash file:
+
+```
+echo 'source <(temporal completion bash)' >>~/.bashrc
+source ~/.bashrc
+```
+
+#### macOS installation
+
+Install `bash-completion` through Homebrew:
+`brew install bash-completion@2`
+
+Add the provided code snippet to your `~/.bashrc` file:
+
+```sh
+[[ -r "/usr/local/etc/profile.d/bash_completion.sh" ]] && . "/usr/local/etc/profile.d/bash_completion.sh"
+```
+
+Verify that `bash-completion` is installed by running `type _init_completion`.
+
+Enable completion for Temporal by adding the following code to your bash file:
+
+```
+echo 'source <(temporal completion bash)' >> ~/.bashrc
+source ~/.bashrc
+```
+
+## Development
+
+The Temporal CLI can be compiled and executed from your system.
+
+To compile the source code into an executable, run:
+
+```bash
+go build -o dist/temporal ./cmd/temporal
+```
+
+To compile the documentation, run:
+
+```bash
+go build -o dist/temporal-docgen ./cmd/temporal-docgen
+```
+
+To run all tests:
+
+```bash
+go test ./...
+```
+
+## Known Issues
+
+- When consuming Temporal as a library in go mod, you may want to replace grpc-gateway with a fork to address URL escaping issue in UI. See <https://github.com/temporalio/temporalite/pull/118>
+
+- When running the executables from the Releases page in macOS you will want to allowlist `temporal` binary in `Security & Privacy` settings:
```

Comment:
Probably don't need the known issues here either.
We don't support consuming the CLI as a library and we don't direct users to install the binary from our releases.
@feedmeapples is this still an issue BTW? Do we need to sign the darwin release somehow?